### PR TITLE
Initial draft of xunit template.

### PR DESCRIPTION
**Pull request by codito**

### CODE REVIEW

**Comment by eerhardt** [comment]
@natemcmaster - I believe you were looking for this right?  You should be able to use this template to use 'dotnet test3' and xunit.

**Comment by natemcmaster** [comment]
@eerhardt yes, thanks. I assume this requires a frankenbuild of xunit.runner.msbuild: the builds on nuget and https://myget.org/gallery/xunit don't have a netcoreapp version.

**Comment by codito** [comment]
@natemcmaster xunit adapter is a build of https://github.com/xunit/visualstudio.xunit/pull/88.
Attaching it here: [xunit.runner.visualstudio.2.2.0-beta4-build1188.nupkg.zip](https://github.com/dotnet/sdk/files/556483/xunit.runner.visualstudio.2.2.0-beta4-build1188.nupkg.zip)

**Comment by natemcmaster** [comment]
@codito thanks. I'm trying to make this template work, but "dotnet test3" appears to hang indefinitely on "Starting test execution, please wait...".

**Review by srivatsn — APPROVED** [review]

**Comment by livarcocc** [comment]
@codito I am hitting the same thing. It hangs when test execution starts.

File: src/Templates/ProjectTemplates/CSharp/.NETCore/CSharpXUnitTest/CSharpXUnitTest.csproj
Comment (bradwilson) [inline src/Templates/ProjectTemplates/CSharp/.NETCore/CSharpXUnitTest/CSharpXUnitTest.csproj]: If this is for .NET Core, why is it targeting desktop CLR?

Comment (srivatsn) [inline src/Templates/ProjectTemplates/CSharp/.NETCore/CSharpXUnitTest/CSharpXUnitTest.csproj]: This project builds the template (on windows). The actual template Is https://github.com/codito/dotnet-sdk/blob/8ea2957a740f419b6df7213090eb2d65b1a48aec/src/Templates/ProjectTemplates/CSharp/.NETCore/CSharpXUnitTest/ProjectTemplate.csproj

**Review by singhsarab — APPROVED** [review]
LGTM 👍

**Comment by srivatsn** [comment]
It looks like the hang has been resolved but there's no output? Waiting for @livarcocc to verify and signoff before merging this in. Tagging @MattGertz for ask mode approval.

**Comment by MattGertz** [comment]
Meets the bar; @srivatsn I'll proactively approve with the understanding that PR will not be initiated until Livar signs off.

**Comment by codito** [comment]
With latest bits from dotnet-cli/rel/1.0.0, tests are running on Windows. Livar is facing an issue with OSX, we're looking into it.

**Comment by codito** [comment]
Fix for the OSX issue is available in https://github.com/dotnet/cli/pull/4558.

**Review by livarcocc — APPROVED** [review]
The issue we were hitting in the CLI was fixed by taking a new version of the testplatform. No changes to the template were required.

**Comment by MattGertz** [comment]
Approved.

**Pull request merged**
